### ISSUE TITLE
False positive `used-before-assignment` in match-case

### ISSUE THREAD

**Issue by tushar-deepsource** [PyCQA/pylint#7260]
### Bug description

```python
foo = "abcd"

match foo:
    case x if "a" in x:
        ...
```

### Command used

```shell
pylint asd.py
```

### Pylint output

```shell
************* Module asd
asd.py:4:21: E0601: Using variable 'x' before assignment (used-before-assignment)
```

### Expected behavior

No issue raised

### Pylint version

```shell
pylint 2.13.7
astroid 2.11.3
Python 3.10.5 (main, Jun 23 2022, 17:14:57) [Clang 13.1.6 (clang-1316.0.21.2.5)]
```

**Comment by jacobtylerwalls** [comment]
Duplicate of #5327
